qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -1,16 +1,143 @@
 import logging
+import numpy as np
 from functools import partial
 from typing import Optional, Any, Tuple
 
 from qcodes import VisaInstrument, Instrument
 from qcodes import ChannelList, InstrumentChannel
 from qcodes.utils import validators as vals
-import numpy as np
-from qcodes.instrument.parameter import MultiParameter, ArrayParameter, ParamRawDataType
+from qcodes.instrument.parameter import MultiParameter, ArrayParameter, \
+    ParamRawDataType
+from qcodes.utils.deprecate import deprecate
+
 
 log = logging.getLogger(__name__)
 
 
+class FixedFrequencyTraceIQ(MultiParameter):
+    """
+    Sweep that returns the real (I) and imaginary (Q) parts of the VNA response.
+    Requires the use of the sweep type to be set to continuous wave mode.
+    See (https://www.rohde-schwarz.com/webhelp/ZNB_ZNBT_HTML_UserManual_en
+    /ZNB_ZNBT_HTML_UserManual_en.htm) under GUI reference -> sweep softtool
+    -> sweep type tab -> CW mode
+    """
+
+    def __init__(self, name: str, instrument: "ZNBChannel",
+                 npts: int, bandwidth: int, channel: int,
+                 check_cw_sweep_first: bool = True) -> None:
+        super().__init__(name, names=("", ""), shapes=((), ()))
+        self._instrument = instrument
+        self.set_cw_sweep(npts, bandwidth)
+        self._channel = channel
```

Comment:
i think it is indeed unused, but then this holds also for the existing frequencysweep parameter of the VNA. I copied this in case it was used somewhere that was unclear. We can try to remove it and see.

Reply:
> in case it was used somewhere that was unclear

would the following approach work better: see if it is used for the case of this parameter, and based on that decide to keep it or not, and leave the existing frequencysweep parameter "as is"?

-------

copy-paste-driver-development is fine, as long as after the copy-paste the pasted code is actually refined, adapted, etc ;)

Reply:
1288fef20562c682f58fce256d64e976165ab054